packages/frontend/src/utils/rateLimit.ts

```diff
@@ -0,0 +1,21 @@
+import { LRUCache } from 'lru-cache'
+import { NextResponse } from 'next/server'
+
+const tokenCache = new LRUCache<string, number>({
+  max: 1000, // 1000 users per min
+  ttl: 60000, // 1min
+})
+
+export default function rateLimit(_res: NextResponse, limit: number, token: string) {
+  const tokenCount = (tokenCache.get(token) || 0) + 1
+  tokenCache.set(token, tokenCount)
+
+  const currentUsage = tokenCount
+  const isRateLimited = currentUsage >= limit
```

Comment:
I believe it should be `currentUsage > limit` as the currentUsage is `value from cache + 1`. Example:
1. limit is 2 -> we should be able to request nonce twice
2. first request: currentUsage = 0 (from tokenCache) + 1 = 1
    - 1 >= limit -> false (first request)
3. second request: currentUsage = 1 (fromTokenCache) + 1 = 2
    - 2 >= limit -> true  -> it's rate limited -> error, which means that we can request only once